tests/Fixtures/TestBundle/Document/Payment.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Tests\Fixtures\TestBundle\Document;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Controller\Payment\VoidPaymentAction;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+/**
+ * @ODM\Document
+ *
+ * @ApiResource(
+ *     itemOperations={
+ *         "get",
+ *         "post_void"={
+ *             "method"="POST",
+ *             "path"="/payments/{id}/void",
+ *             "controller"=VoidPaymentAction::class,
+ *             "deserialize"=false,
+ *         },
+ *     },
+ * )
+ */
+class Payment
+{
+    /**
+     * @var int|null
+     *
+     * @ODM\Id(strategy="INCREMENT", type="integer")
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $amount;
```

Comment:
Not mapped as Doctrine MongoDB ODM does not have a decimal type. I'm not sure what's the right thing to do here... If I use `type="float"`, the type validation fails.

Reply:
https://github.com/doctrine/mongodb-odm/issues/1835